lib/atat-web-api-stack.ts

```diff
@@ -85,5 +88,26 @@ export class AtatWebApiStack extends cdk.Stack {
       provisioningSfn.provisioningQueueConsumer.method,
       new apigw.LambdaIntegration(provisioningSfn.provisioningQueueConsumer.fn)
     );
+
+    const atatIdp = new idp.CognitoIdentityProvider(this, "AtatIdp", {
+      domainPrefix: `atat${environmentName.toLowerCase()}`,
+      scopes: [
+        {
+          resourceServerName: "atat",
+          scopes: [
+            { name: "read-cost", description: "Allow reading cost information" },
+            { name: "read-portfolio", description: "Allow reading information about portfolios" },
+            { name: "write-portfolio", description: "Allow creating and updating portfolios" },
+          ],
+        },
```

Comment:
Should these `scopes` live in another file?

Reply:
Maybe? We probably want to narrow down exactly what to call our Resource Server. These likely _should_ be context values actually.